### PR TITLE
MM-T614 Search on Outgoing Webhooks page

### PR DESCRIPTION
**Pull request by MikeworX**
**_Summary_**

This PR validates that the search functionality on the Integrations>Outgoing Web-hooks page is working as intended.

**_Steps_**

- 3 outgoing web-hooks are created, each with different names and trigger words
- asserts  are done on names, case, partial, common name, special character, no results, to insure expected web-hooks are displayed (or not)

This test requires webhook_serve.js to be running.

Screen Shot 2020-10-21 at 12 39 48 PM

**_Links_**

[vercel](https://automation-test-cases.vercel.app/test/MM-T614)
[TM4J](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testCase/MM-T614)

### CODE REVIEW

File: e2e/cypress/integration/integrations/outgoing_webhook/search_on_outoing_webhooks_spec.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            });
+        });
+    });
+
+    it('MM-T614 Search on Outgoing Webhooks page', () => {
+        // * Assert that search for Alpha (lower-case) returns only Alpha webhook
+        cy.visit(`/${testTeam}/integrations/outgoing_webhooks`);
+        cy.get('#searchInput').type('alpha').then(() => {
+            cy.get('.backstage-list__item').should('not.contain', triggerB).should('not.contain', triggerC).contains(triggerA);
+        });
```
Comment (saturninoabril) [inline e2e/cypress/integration/integrations/outgoing_webhook/search_on_outoing_webhooks_spec.js:87]: No need to do `.then`. I suggests,
```suggestion
        cy.get('#searchInput').type('alpha');
        cy.get('.backstage-list__item').should('not.contain', triggerB).should('not.contain', triggerC).contains(triggerA);
```

File: e2e/cypress/integration/integrations/outgoing_webhook/search_on_outoing_webhooks_spec.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+        // * Assert that search for Bravo (upper-case) returns only Bravo webhook
+        cy.visit(`/${testTeam}/integrations/outgoing_webhooks`);
+        cy.get('#searchInput').type('BRAVO').then(() => {
+            cy.get('.backstage-list__item').should('not.contain', triggerA).should('not.contain', triggerC).contains(triggerB);
+        });
```
Comment (saturninoabril) [inline e2e/cypress/integration/integrations/outgoing_webhook/search_on_outoing_webhooks_spec.js:93]: Revisiting the page multiple times will make the test longer. I suggest to just clear the input then type again.
```suggestion
        cy.get('#searchInput').clear().type('BRAVO');
        cy.get('.backstage-list__item').should('not.contain', triggerA).should('not.contain', triggerC).contains(triggerB);
```

Same comment below.

**Review by saturninoabril — CHANGES_REQUESTED** [review]
@MikeworX Thanks. Test flow looks good, except for few comments.

File: e2e/cypress/integration/integrations/outgoing_webhook/search_on_outoing_webhooks_spec.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
```
Comment (saturninoabril) [inline e2e/cypress/integration/integrations/outgoing_webhook/search_on_outoing_webhooks_spec.js:1]: Please remove this file.

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @MikeworX for updating. Tested and passed, except for a request to remove old file. I'm all good after that.

**Comment by MikeworX** [comment]
Ah - thanks - missed that. I forgot that rename ≠remove.

**Review by saturninoabril — APPROVED** [review]
Thanks @MikeworX! LGTM, tested and passed.

**Review by catalintomai — APPROVED** [review]

**Comment by jfrerich** [comment]
@prapti, gentle reminder for review.

**Review by prapti — APPROVED** [review]
LGTM!

**Pull request merged**